integration-tests/init.spec.js

```diff
@@ -1,107 +1,149 @@
+          it('should initialize the tracer, if DD_INJECT_FORCE', () => doTestForced('true\n', ...telemetryForced))
+        })
+        context('with debug', () => {
+          useEnv({ DD_TRACE_DEBUG })
+
+          it('should not initialize the tracer', () =>
+            doTest(`Aborting application instrumentation due to incompatible_runtime.
+Found incompatible runtime nodejs 3.0.0, Supported runtimes: nodejs >=18.
+false
+`, ...telemetryAbort))
+          it('should initialize the tracer, if DD_INJECT_FORCE', () =>
+            doTestForced(`Aborting application instrumentation due to incompatible_runtime.
+Found incompatible runtime nodejs 3.0.0, Supported runtimes: nodejs >=18.
+DD_INJECT_FORCE enabled, allowing unsupported runtimes and continuing.
+Application instrumentation bootstrapping complete
+true
+`, ...telemetryForced))
+        })
+      })
     })
-    it(`should ${esmWorks ? '' : 'not '}initialize ESM instrumentation, if DD_INJECTION_ENABLED`, () => {
-      return runTest(cwd, 'init/instrument.mjs', { NODE_OPTIONS, DD_INJECTION_ENABLED }, `${esmWorks}\n`)
+    context('when node version is more than engines field', () => {
+      useEnv({ FAKE_VERSION: '50000.0.0', NODE_OPTIONS })
+
+      it('should initialize the tracer, if no DD_INJECTION_ENABLED', () => doTest('true\n'))
+      context('with DD_INJECTION_ENABLED', () => {
+        useEnv({ DD_INJECTION_ENABLED })
+
+        context('without debug', () => {
+          it('should initialize the tracer', () => doTest('true\n', ...telemetryGood))
+          it('should initialize the tracer, if DD_INJECT_FORCE', () =>
+            doTestForced('true\n', ...telemetryGood))
+        })
+        context('with debug', () => {
+          useEnv({ DD_TRACE_DEBUG })
+
+          it('should initialize the tracer', () =>
+            doTest('Application instrumentation bootstrapping complete\ntrue\n', ...telemetryGood))
+          it('should initialize the tracer, if DD_INJECT_FORCE', () =>
+            doTestForced('Application instrumentation bootstrapping complete\ntrue\n', ...telemetryGood))
+        })
+      })
     })
   })
 }
 
 describe('init.js', () => {
-  before(async () => {
-    sandbox = await createSandbox()
-    cwd = sandbox.folder
-  })
-  afterEach(() => {
-    proc && proc.kill()
-  })
-  after(() => {
-    return sandbox.remove()
-  })
+  useSandbox()
 
   testInjectionScenarios('require', 'init.js', false)
+  testRuntimeVersionChecks('require', 'init.js')
 })
 
 describe('initialize.mjs', () => {
-  before(async () => {
-    sandbox = await createSandbox()
-    cwd = sandbox.folder
-  })
-  afterEach(() => {
-    proc && proc.kill()
-  })
-  after(() => {
-    return sandbox.remove()
-  })
+  useSandbox()
 
   context('as --loader', () => {
     testInjectionScenarios('loader', 'initialize.mjs', true)
+    // TODO uncomment next line and fix this before enabling ESM in SSI
```

Comment:
Is this even going to be supported at all? And if yes, we should have a test right away even if it doesn't work to at least ensure it doesn't work as expected.

Reply:
As it turns out, by switching to using non-faked Node.js versions to address your previous comment, these tests started passing. The faked out `setversion.js` wasn't working in ESM-land.